models/orogen/logger.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "syskit/network_generation/logger"
+
+# OroGen.logger already exists, we can't access the model for logger::Logger
+# using the method syntax
+logger_m = OroGen.syskit_model_by_orogen_name("logger::Logger")
+
+Syskit.extend_model logger_m do
+    provides Syskit::LoggerService
+
+    # Customizes the configuration step.
+    #
+    # The orocos task is available from orocos_task
+    #
+    # The call to super here applies the configuration on the orocos task. If
+    # you need to override properties, do it afterwards
+    #
+    # def configure
+    #     super
+    # end
+
+    def update_properties
+        super
+    end
+
+    def rotate_log
+        previous_file = orocos_task.property("file").read
```

Comment:
Do not access `orocos_task` to read properties. Just use `properties.<property_name>`

Reply:
?

_Never_ access orocos_task in runtime code when using Syskit.

We (unfortunately) still have to do it for component operations, but they are very rarely used. `<task>.properties` gives access to properties. Since `setup_default_logger` is now a syskit-specific implementation, it can be changed to be compatible with that too.